File: Harish Kumar/Graph/Max Area of Island/Max Area of Island.java
class Solution {
    public static int islandFind(int grid[][], int i, int j)
    {
        if(i < 0 || j < 0 || i >= grid.length || j >= grid[0].length || grid[i][j] == 0  )
        {
            return 0;
        }
        grid[i][j] = 0;
        
        return 1+(islandFind(grid, i-1, j) +  islandFind(grid, i, j+1) + islandFind(grid, i, j-1) + islandFind(grid, i+1, j)) ;
        
      
    }
    public int maxAreaOfIsland(int[][] grid) {
        
        // boolean vis[][] = new boolean[grid.length][grid[0].length];
        int maxArea = 0;
        
        for(int i = 0; i < grid.length;i++ )
        {
            for(int j = 0; j < grid[0].length;j++)
            {
                if(grid[i][j] == 1 )
                {
                   int max =  islandFind(grid,i,j);
                    maxArea = Math.max(max,maxArea);
                }
            }
        }
        return maxArea;
    }
}

File: Shreya/Graphs/Max Area of Island.cpp
class Solution {
public:
    int dfs(vector<vector<int>>& grid, int i, int j)
    {
        if(i<0||i>=grid.size()||j<0||j>=grid[0].size())
        {
            return 0;
        }
        if(grid[i][j]==0)
        {
            return 0;
        }
        grid[i][j]=0;
        return 1 + dfs(grid, i+1, j) + dfs(grid, i-1, j) + dfs(grid, i, j-1) + dfs(grid, i, j+1);
    }
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int maxislands{};
        for(int i=0;i<grid.size();i++)
        {
            for(int j=0;j<grid[0].size();j++)
            {
                if (grid[i][j]==1)
                {
                    maxislands = max(maxislands,dfs(grid,i,j));
                }
            }
        }
        return maxislands;
    }
};